docs/how-to/DataScienceScripts.md

```diff
@@ -0,0 +1,154 @@
+
+    ```python
+    train_model_cmp = gl_pipeline_components[0](
+        dataset_folder=pipeline_job_input,
+    )
+    predict_model_cmp = gl_pipeline_components[1](
+        model_folder=train_model_cmp.outputs.model_folder,
+        dataset_folder=pipeline_job_input,
+    )
+    score_model_cmp = gl_pipeline_components[2](
+        pred_folder=predict_model_cmp.outputs.pred_folder,
+    )
+    ```
+
+### Data Science Script Configuration
+
+The Data Science script must have configuration, which defines the configuration parameter required for the data science scripts. The configuration file should be in yaml format and should be easily configurable. For example, the configuration file for the nyc_taxi model will be stored in the `src/nyc_taxi/common/nyc_config.yaml` folder. And the DS scripts can read the configuration file and use the parameters defined in it.  Below is an example of how the `nyc_taxi_config.yaml` file looks like:
+
+    ```yaml
+    dry_run: True
+    train:
+       constant_history_window: 32
+       gradient_accumulation_steps: 40
+       batch_size: 12
+    ........
+    ```
+
+    And the `train.py` script will read the configuration file as follows:
+
+    ```python
+    # Load NYC yaml configuration
+    import yaml
+    nyc_config = yaml.safe_load(open(args.nyc_taxi_config))
+    dry_run = nyc_config['dry_run']
+    train_cfg = nyc_config["train"]
+    gradient_accumulation_steps = train_cfg["gradient_accumulation_steps"]
+    ```
+
+### Data Science Python Code Dependencies
+
+The Data Science script must define the python dependencies required to run the script. The dependencies can be defined in a requirements.txt file or in a conda environment file.
+
+    Below is example of how the `mlops/nyc_taxi/environment/conda.yml` will look like:
+
+    ```yaml
+    name: prs-env
+    channels:
+    - conda-forge
+    dependencies:
+    - python=3.9
+    - pip
+    - pip:
+        - python-dotenv
+        - pandas
+        - numpy==1.23.5
+        - scikit-learn==1.3.2
+        - mlflow>=2.9.2
+        - azureml-mlflow>=1.5
+    ```
+
+### Data Science Script Observability
+
+The Data Science script must have logging enabled to track the metrics for observability purposes. The logging must be done in a structured way such that it can be easily monitored.
+
+Machine Learning model scripts, for example train.py, predict.py, etc. will publish parameters using MLFlow to ease debugging and troubleshooting. Below is an example of how the parameters can be logged using MLFlow:
+
+    ```python
+    import argparse
+    import mlflow
+
+    args = parser.parse_args()
+    mlflow.log_params(vars(args))
+    ```
+
+### Data Science Script Testing
+
+The Data Science script must be tested thoroughly to ensure that it works as expected. The testing must cover all the components of the script and should be automated as part of the CI pipeline to catch errors prior to merging code into `main`.
```

Comment:
We would not want automated tests to run as part of CI in the prototyping repo.  If we run them, it should be in the model factory during promotion.  The only tests in the prototyping would be the inference tests which are promoted alongside the model.  These however, would not be run as part of CI.

Reply:
Thanks @marshallbentley changed it